### PR TITLE
Discovering project ID from the local metadata service

### PR DESCRIPTION
**Pull request by hiranya911**
Support for auto discovering the project ID when running on Compute Engine, Cloud Run etc.

Verified by running a build on a Compute Engine instance.

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -56,14 +56,15 @@ export function addReadonlyGetter(obj: object, prop: string, value: any): void {
 }
 
 /**
- * Determines the Google Cloud project ID associated with a Firebase app by examining
- * the Firebase app options, credentials and the local environment in that order.
+ * Returns the Google Cloud project ID associated with a Firebase app, if its explicitly
```
Comment (bojeil-google) [inline src/utils/index.ts]: it's

File: test/unit/auth/credential.spec.ts [hunk 2]
```diff
@@ -328,6 +329,47 @@ describe('Credential', () => {
         });
       });
     });
+
+    it('should cache discovered project id', () => {
+      const expected = 'test-project-id';
```
Comment (bojeil-google) [inline test/unit/auth/credential.spec.ts]: nit: Rename `expected` to `expectedProjectId` here and below.

File: src/utils/index.ts (hunk 1 above)
Comment (hiranya911) [reply src/utils/index.ts]: Done

File: test/unit/auth/credential.spec.ts (hunk 2 above)
Comment (hiranya911) [reply test/unit/auth/credential.spec.ts]: Done

**Comment by hiranya911** [comment]
@bojeil-google I've made the suggested changes.

**Pull request merged**